Skip automigrating for existing tables

lpoli opened the pull request:
### Fixes
This PR add version table so that we can later on migrate tables and its data based on this information.
Also, the automigration will drop tables only if `automigration: true`, default value is `false`.

lpoli committed:
Skip automigrating for existing tables

cnlangzi commented:
I don't how it works if a table is configured as keep_tables on yaml file, and there are new columns should be added, or some old columns should be dropped. let's waiting for @peterlimg  to review

lpoli committed:
Set default automigration value to false
Drop all tables and automigrate schema only if automigration is set to true or database is newly created

lpoli committed:
Add version info of blobber
Version info can be used in comming version to atleast effectively migrate database

lpoli committed:
Remove reduntant check

lpoli committed:
Merge branch 'staging' into fix/automigrate-existing-table-issue

lpoli committed:
Always automigrate without deleting data

cnlangzi commented on code/go/0chain.net/blobbercore/automigration/automigration.go:
```diff
@@ -129,20 +134,9 @@ func grantPrivileges(db *gorm.DB) error {
-			continue
-		}
-
 		err := db.Migrator().DropTable(tableName)
```
We need `Migration`. `drop` and `create` is `recreate` with fresh data, is not `schema migration`.

cnlangzi commented on config/0chain_blobber.yaml:
```diff
@@ -97,18 +97,12 @@ pg:
   password: postgres
 db:
-  automigrate: true # true to automigrate with gorm's Automigration
+  automigrate: false # true to automigrate with gorm's Automigration
```
it is unnecessary. AutoMigrate should always be checked. gorm will compare schema between Models and database, and build upgrade scripts for new tables, new columns, changed columns.....

AutoMigrate is `migrate schema without losing data`.

lpoli replied:
@cnlangzi you got stale view of the PR.
I had already made the changes that you have requested.

cnlangzi replied:
sorry, github view doesn't refresh properly.

cnlangzi reviewed (changes requested):
I think you have some misunderstand about `AutoMigrate` on schema. why a table has to be dropped before migrating its schema?  nobody will drop table to loss all data when it is migrated. I can't understand your idea.

cnlangzi commented on code/go/0chain.net/blobber/config.go:
```diff
@@ -132,6 +131,10 @@ func setupConfig(configDir string, deploymentMode int) {
 	config.Configuration.WebsiteUrl = viper.GetString("website_url")
 	config.Configuration.LogoUrl = viper.GetString("logo_url")
 	config.Configuration.Description = viper.GetString("description")
+	config.Configuration.Version = viper.GetString("version")
```
add `version` in yaml file, it will have the same issue with keep_tables. it still has a deployment issue for blobber owner.

lpoli committed:
Remove version info

cnlangzi commented on config/0chain_blobber.yaml:
```diff
@@ -1,4 +1,4 @@
-version: 1.0
+version: "1.0"
```
it is configured for nothing. it is better to remove it for now.

cnlangzi reviewed (approved):
LGTM

service-0chain commented:
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/automigrate-existing-table-issue  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

cnlangzi committed:
Merge branch 'staging' into fix/automigrate-existing-table-issue
